aidial_adapter_bedrock/dial_api/storage.py

```diff
@@ -1,34 +1,56 @@
 import base64
 import hashlib
 import io
-from typing import TypedDict
+import re
+from typing import List, Mapping, Optional, TypedDict
 
 import aiohttp
 
+from aidial_adapter_bedrock.dial_api.auth import Auth
+from aidial_adapter_bedrock.utils.env import get_env, get_env_bool
 from aidial_adapter_bedrock.utils.log_config import bedrock_logger as log
 
 
 class FileMetadata(TypedDict):
     name: str
+    parentPath: str
+    bucket: str
+    url: str
     type: str
-    path: str
+
     contentLength: int
     contentType: str
 
 
 class FileStorage:
-    base_url: str
-    api_key: str
+    dial_url: str
+    bucket_url: str
+    upload_url: str
+    auth: Auth
+
+    def __init__(self, dial_url: str, upload_dir: str, bucket: str, auth: Auth):
+        self.dial_url = dial_url
+        self.bucket_url = f"{self.dial_url}/v1/files/{bucket}"
+        self.upload_url = f"{self.bucket_url}/{upload_dir}"
+        self.auth = auth
 
-    def __init__(self, dial_url: str, base_dir: str, api_key: str):
-        self.base_url = f"{dial_url}/v1/files/{base_dir}"
-        self.api_key = api_key
+    @classmethod
+    async def create(cls, dial_url: str, base_dir: str, auth: Auth):
+        bucket = await FileStorage._get_bucket(dial_url, auth)
+        return cls(dial_url, base_dir, bucket, auth)
 
-    def auth_headers(self) -> dict[str, str]:
-        return {"api-key": self.api_key}
+    @staticmethod
+    async def _get_bucket(dial_url: str, auth: Auth) -> str:
+        async with aiohttp.ClientSession() as session:
```

Comment:
Can we use one session for get and put requests?

Reply:
Indeed, fixed.